### ISSUE TITLE
Strange behavior (?)

### ISSUE THREAD

**Issue by joqb** [ogrecio/RanBoost#1]
Dear Oscar,

I experimented a little bit with RanBOOST and I find it very interesting and very easy to set up.
However, I think that I experienced some potentially weird behavior. 
![RanBoost_run](https://user-images.githubusercontent.com/11645116/222124933-cb835d92-4308-44b3-b87f-3f2493c1559b.png)
According to the screen output, I would say that convergence didn't happen yet: 
at round 20000, the optimal iteration is 19933 with an MSE of 88.313.
Strangely, in the file MSE_test, MSE values are around 820 for iteration 19933 and as you can see in the plot below, MSE values are first increasing rapidly, then decreasing until about 740 at around 4000 iterations, and finally drifting towards higher values.
![RanBoost_MSE_test](https://user-images.githubusercontent.com/11645116/222090596-baec8f7e-716c-40e0-b44b-324d8f22cf37.png)

Why are the values not the same between screen output and file (88.313 vs. 822.88)?
Why is it not the iteration with an MSE of ca. 740 that is defined as optimal iteration?
Is the program sensitive to high values for the number of iterations, so that after passing a local optima, MSE values are drifting upwards?

Best regards,
Jonathan

**Comment by ogrecio** [comment]
Dear Jonathan, 
thank you for posting this. In order to understand your query, please confirm that the MSE in the plot is from the MSE_test file. In that case, it does not need to be the same as the MSE in the tuning set shown in the screen (88.313). They are referring to different sets. The program does not use the testing set to stop the algorithm at all.  Otherwise, some overfitting would be created. The testing set is a completely independent data set. This would explain the differences.

The MSE plot in the testing set shows a slight overfitting, though. What value for the smoothing parameter (v) did you use? Maybe you could try increasing it, as the convergence appears to be too slow.
Please let me know if this is clear.
Kind regards,
Oscar

**Comment by joqb** [comment]
Dear Oscar,

I confirm that the MSE in the plot is from the MSE_test file. 
So, if I understand well, MSE_tune is the MSE of a random individual set of the training population.
For this test, I used 0.2 as smoothing parameter. Using 0.3, 0.4, and 0.5 as smoothing parameter, the analyses converged after 9226, 1431, and 2647 iterations, respectively.

Is it possible that, for a certain combination of mtry and smoothing parameters, the analysis never converges ? +

Best regards,
Jonathan

**Comment by ogrecio** [comment]
Correct. MSE_tune is from a random set from the training population.

It is possible that the analyses never converges. but it is quite implausible.
Kind regards,
